src/path.rs

```diff
@@ -189,6 +189,48 @@ impl PathBuf {
     pub fn display(&self) -> BytesDisplay {
         BytesDisplay(&self.0)
     }
+
+    /// Append to the path.
+    ///
+    /// This will add a separator if needed. Note that if the argument
+    /// is an absolute path, `self` will be replaced with that path.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the argument is not a valid path.
+    #[track_caller]
+    pub fn push(&mut self, p: impl AsRef<[u8]>) {
+        #[track_caller]
+        fn inner(this: &mut PathBuf, p: &[u8]) {
+            // Panic if the arg is not a valid path.
+            let p = Path::try_from(p).expect("invalid path arg");
```

Comment:
Nit: https://doc.rust-lang.org/std/option/enum.Option.html#recommended-message-style

Reply:
Updated to "push arg must be a valid path". Their recommendation to use "should" seems weird to me, "must" feels more accurate.